rust-runtime/inlineable/src/hyper_body_wrap_stream.rs

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_http::byte_stream::error::Error as ByteStreamError;
+use aws_smithy_http::byte_stream::ByteStream;
+use aws_smithy_http::event_stream::MessageStreamAdapter;
+use aws_smithy_http::result::SdkError;
+use bytes::Bytes;
+use futures_core::stream::Stream;
+use std::error::Error as StdError;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pub(crate) struct HyperBodyWrapEventStream<T, E>(MessageStreamAdapter<T, E>);
```

Comment:
nit: I'd prefer it if this were not an inlineable and instead lived in a runtime crate, `#[doc(hidden)]`. We can add the docs we've sprinkled in the code generator as proper Rust docs, contributors have a better development experience, and each user won't have to pay the cost of running the below tests when compiling their generated SDK.

Reply:
Having `HyperBodyWrapEventStream` in a runtime crate might help solve https://github.com/awslabs/smithy-rs/pull/2910#discussion_r1306191209 as well. I think it's worth incorporating this as part of the PR.

Reply:
In addition to what you pointed out, moving this to a runtime crate also helped address https://github.com/awslabs/smithy-rs/pull/2910#discussion_r1306191209. Thanks for the suggestion. Addressed in f5c8a96.